src/EventTeachingService.cpp

```diff
@@ -39,6 +39,24 @@ Processed EventTeachingService::handleMessage(unsigned int opc, CANFrame *msg)
 
   switch (opc) 
   {
+    case OPC_MODE:
+      // 76 - Set Operating Mode
+      //DEBUG_SERIAL << F("> MODE -- request op-code received for NN = ") << nn << endl;
+      byte requestedMode = msg->data[3];
```

Comment:
request mode is visible in the whole switch statement. You can put the code for case OPC_MODE in {...}. 
But looking at other cases, shouldn't there be a test for NORMAL mode? And that we have a node number?
You can only change to or from LEARN mode if you are in NOMRAL mode and have an assigned node number.
If you add such an if-statement then you automatically get a {...} block that limits the visibility of requestedMode.

Reply:
We don't need to test for Normal mode because if it isn't in Normal mode MNS will have shut it down.

Reply:
Sorry don't understand "promulgation". Assuming you mean that MNS stops other services from handling the MODE code. 
This relies on the "return PROCESSED" in MNS which I commented on elsewhere.

Reply:
My understanding was that the services were offered the opcode in sequence.  Since MNS is service 1, it would always be able to do the initial checking of the opcode for length and node number.  If the service, MNS or later, uses the opcode then it will "return PROCESSED" at which point there is no longer a message to offer to subsequent services and the "for" loop exits.

Reply:
That makes sense if MNS is actually the first service in the list. How can we ensure that MNS is actually the first service in the list? Should we tolerate users who don't put MNS first? (I.e. make the system work somewhat, or should we error and stop?)

Reply:
The question here is does "for (Service * service : services)" run through the list in service number order or the sequence that is in the sketch pre-processor listing "VLCB::Controller controller(........)".  If it is the former, then there should be no issue but if it is the latter there is.

Since MNS is the only mandatory service, I have always been concerned about it being called a service at all as it is as fundamental to function as configuration and control. Is it possible to make its inclusion integral to the library so that the user does not have the option to control when it is called or even to inadvertently leave it out?

Reply:
Yes, that for construct will iterate the service list in order.
Yes, I agree. I have been thinking this. I like to have MNS as an object in the user sketch so that it can be configured. Although we don't have any configurations in MNS at this stage. Just don't want to close the door for the future. 
Anyways, let's leave this for a f2f discussion.

Reply:
Agreed for f2f discussion